src/uattributes/uattributesvalidator.rs

```diff
@@ -634,6 +706,65 @@ mod tests {
         let status = validator.validate(&attributes);
         assert!(status.is_ok() == expected_result);
         if status.is_ok() {
+            assert!(UAttributesValidators::Notification
+                .validator()
+                .validate(&attributes)
+                .is_err());
+            assert!(UAttributesValidators::Request
+                .validator()
+                .validate(&attributes)
+                .is_err());
+            assert!(UAttributesValidators::Response
+                .validator()
+                .validate(&attributes)
+                .is_err());
+        }
+    }
+
+    #[test_case(Some(UUIDBuilder::new().build()), Some(origin()), None, None, false; "fails for missing destination")]
+    #[test_case(Some(UUIDBuilder::new().build()), Some(origin()), Some(destination()), None, true; "succeeds for both topic and destination")]
```

Comment:
"succeeds for both origin and destination"

Reply:
Fixed